src/sdk/namespace_cache.js

```diff
@@ -472,26 +527,39 @@ class NamespaceCache {
                         { md_conditions: params.md_conditions, cache_object_md: params.object_md });
                 }
 
-                this.stats_collector.update_namespace_cache_stats({
+                this.stats_collector.update_cache_stats({
                     bucket_name: params.bucket,
                     miss_count: 1,
                     range_op: true,
                 });
             } else {
                 upload_params.last_modified_time = (new Date(params.object_md.create_time)).getTime();
+                const start = process.hrtime.bigint();
                 _global_cache_uploader.submit_background(
                     params.object_md.size,
                     async () => {
                         const upload_res = await this.namespace_nb.upload_object(upload_params, object_sdk);
-                        this.stats_collector.update_namespace_cache_stats({
+                        // update latency stats on 'end'
+                        upload_res.once('end', () => {
```

Comment:
As noted on other cases, when using `await upload_object` we don't get back a stream and we do no need to wait more, so we can just update the stats directly after await.